src/ls/main.cpp

```diff
@@ -14,8 +14,8 @@
 #include <sys/types.h>
 #include <compat/compat.h>
 #include <ncpp/Direct.hh>
-#ifndef __linux__
-#define AT_NO_AUTOMOUNT 0 // not defined on freebsd
+#ifndef AT_NO_AUTOMOUNT
+#define AT_NO_AUTOMOUNT 0x800
```

Comment:
isn't this going to start sending 0x800 to a FreeBSD kernel? freebsd flags do not overlap with linux ones:

https://www.freebsd.org/cgi/man.cgi?query=fstatat&sektion=2&n=1

also, if we send a newer flag to a kernel that's not ready to accept it, might it not return a fail? i'm with you on the first line's change, but i find the second more dubious.

Reply:
` #define    AT_REMOVEDIR        0x0800    /* Remove directory instead of file */`

Reply:
>  freebsd flags do not overlap with linux ones:

Ah, good catch. I will go back to 0 and re-build and see what happens.

Reply:
Seems to run ok with 0. Added a commit to change it. Lemme know if I should squash it or you just squash on eventual merge.

Reply:
i rebase as you offer them, so right now it would go in as two commits. i enjoy the rich living tapesty of a git history full of fuckups and one-offs. if you'd like it to be a single commit, squash on your end =]. and thanks a lot for this valuable work! if you demonstrate that things work on older environments, feel encouraged to update the minimums listed in the README etc.

Reply:
👍 , fine as it is then